src/Preprocessing.py

```
@@ -0,0 +1,67 @@
+    b, g, r = cv2.split(img)
+    clahe = cv2.createCLAHE(clipLimit=clipLimit, tileGridSize=tileSize)
+    b = clahe.apply(b)
+    g = clahe.apply(g)
+    r = clahe.apply(r)
+    equalized = cv2.merge((b, g, r))
+    blurred = cv2.GaussianBlur(equalized, blur_kernel, sigmaX)
+    return blurred
+
+# Duyệt tất cả ảnh trong folder gốc, xử lý và lưu giữ cấu trúc thư mục
+def process_all_images(input_root, output_root):
+    for root, dirs, files in os.walk(input_root):
```

Ruff (0.11.9)

Loop control variable `dirs` not used within loop body

Rename unused `dirs` to `_dirs`

(B007)